recipes/libsodium/all/conanfile.py

```diff
@@ -87,12 +87,14 @@ def _msvc_sln_folder(self):
                 "14": "vs2015",
                 "15": "vs2017",
                 "16": "vs2019",
+                "17": "vs2022",
             }
         else:
             folder = {
                 "190": "vs2015",
                 "191": "vs2017",
                 "192": "vs2019",
+                "193": "vs2022",
```

Comment:
Have you tried? There is no vs2022 folder in sources, so I don't understand how it could work.

Reply:
Yes, of course I tried it. And it worked just fine locally with 
`conan create .\conanfile.py libsodium/1.0.18@ -s compiler.version=17`

Reply:
I know what's wrong, I've forgotten the return in `_msvc_sln_folder`, therefore `vs2019` folder is always chosen. A

Reply:
Yes it fails now, but that's also partly because I didn't see that 1.0.18 was released 2 years(!!) ago, and the master branch of libsodium has vs2022 support.

Reply:
So in reality all that's needed is the commit you made? Seems so when I run locally, that it upgrades the 2019 project

```libsodium/1.0.18: Calling build()
**********************************************************************
** Visual Studio 2022 Developer Command Prompt v17.0.6
** Copyright (c) 2021 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'

Microsoft Visual Studio 2022 Version 17.0.6.
Copyright (C) Microsoft Corp. All rights reserved.
Upgrading project 'libsodium'...
        Configuration 'ReleaseLTCG|x64': changing Platform Toolset to 'v143' (was 'v142').
        Configuration 'ReleaseLTCG|Win32': changing Platform Toolset to 'v143' (was 'v142').
        Configuration 'DebugLTCG|Win32': changing Platform Toolset to 'v143' (was 'v142').
        Configuration 'DebugLTCG|x64': changing Platform Toolset to 'v143' (was 'v142').
        Configuration 'ReleaseDLL|x64': changing Platform Toolset to 'v143' (was 'v142').
        Configuration 'ReleaseLIB|x64': changing Platform Toolset to 'v143' (was 'v142').
        Configuration 'ReleaseDLL|Win32': changing Platform Toolset to 'v143' (was 'v142').
        Configuration 'ReleaseLIB|Win32': changing Platform Toolset to 'v143' (was 'v142').
        Configuration 'DebugLIB|Win32': changing Platform Toolset to 'v143' (was 'v142').
        Configuration 'DebugDLL|Win32': changing Platform Toolset to 'v143' (was 'v142').
        Configuration 'DebugLIB|x64': changing Platform Toolset to 'v143' (was 'v142').
        Configuration 'DebugDLL|x64': changing Platform Toolset to 'v143' (was 'v142').
```